docs/operations/upgrade-gardener.md

```diff
@@ -0,0 +1,87 @@
+
+### Verify Readiness
+
+Before moving to the next step, you must verify that the `Garden` resource has been successfully updated. Check the following:
+
+- The `gardener.cloud/operation` annotation is removed, indicating the operation is complete.
+- The `.status.gardener.version` field shows your target version.
+- The `.status.observedGeneration` matches the `.metadata.generation`, meaning the latest configuration has been processed.
+- The `.status.lastOperation.state` is `Succeeded`.
+
+At this stage, it's normal for the `gardenlet`s to still be running the old version. This is similar to how Kubernetes upgrades its control plane before the `kubelet`s on worker nodes.
+
+Finally, check the health conditions (`.status.conditions[]`) in the `Garden` resource to ensure they all report `True`.
+
+## 3. Upgrade Your `gardenlet`s and Extensions
+
+Next, upgrade the `gardenlet`s (and optionally your Gardener extensions).
+
+### Unmanaged Seeds
+
+Start with `gardenlet`s that manage "unmanaged" seeds - seed clusters that are not created via `Shoot`s. These should be configured using `Gardenlet` resources in the `garden` namespace.
+
+If you've enabled auto-updates (by adding the `operator.gardener.cloud/auto-update-gardenlet-helm-chart-ref=true` label to your `Gardenlet` resources), the `gardener-operator` will automatically trigger the upgrade. The `gardenlet`s will then perform a self-upgrade.
+
+Alternatively, if you manage your `Gardenlet` resources with a GitOps tool like Flux, you should not use the auto-update label. Instead, update the Helm chart reference (`.spec.deployment.helm.ociRepository.ref`) in your configuration and apply the changes yourself.
+
+### Managed Seeds
+
+Once a `gardenlet` is upgraded, it automatically begins upgrading any "managed" seeds it controls. No manual action is needed for this step. Note that a `ManagedSeed` upgrade will wait until any ongoing reconciliation of its underlying `Shoot` cluster is complete.
+
+To prevent overloading the system, these upgrades are staggered using a "jitter" period, so they won't all start at once. By default, this period is 5 minutes. You can adjust it in the `gardenlet`'s [component configuration](../../example/20-componentconfig-gardenlet.yaml) by setting `.controllers.managedSeed.syncJitterPeriod`. Set it to `0` to start all upgrades immediately, or increase it if you have many seed clusters to manage the load.
+
+### Extensions
+
+You can upgrade extensions by updating the version in their `ControllerDeployment` resource. This can be done at the same time as the `gardenlet` upgrade, but always check the extension's release notes to ensure compatibility.
```

Comment:
If an extension is installed through the `extension.operator.gardener.cloud` API, such a change would be rolled back.

How about something along these lines?

```suggestion
To upgrade extensions managed by the `gardener-operator`, update the OCI reference in the `extension.operator.gardener.cloud` resource located within the runtime cluster.

Extensions, that were manually deployed to the virtual Garden, update the version in their `ControllerDeployment` resource. 

This can be done at the same time as the `gardenlet` upgrade, but always check the extension's release notes to ensure compatibility.
```